test/library/packages/UnitTest/UnitTest.chpl

```diff
@@ -1052,16 +1053,15 @@ module UnitTest {
     catch e: DependencyFound {
       var allTestsRan = true;
       for superTest in testObject.testDependsOn {
-        var checkCircleStatus = checkCircle.find(superTest: string);
-        // cycle is checked
-        if checkCircleStatus[1]{
+        try {
+          var idx = checkCircle.indexOf(superTest:string);
           testsSkipped[testName] = true;
           circleFound = true;
           var failReason = testName + " skipped as circular dependency found";
           testResult.addSkip(testName, failReason);
           testStatus[testName] = true;
           return;
-        }
+        } catch {}
```

Comment:
I think you have to catch a particular error from `indexOf` here.

Reply:
yeah, please never write `catch {}`. Even if that's what you think is right, put a comment explaining why the error can be safely ignored.

Reply:
This is really odd, but when I rewrite the try/catch to something like...

```
try { ... } catch e: IllegalArgumentError {} catch e { throw e; }
```

The `UnitTest` module tests fail with...

```
./UnitTest.chpl:1064: internal error: did not find the insertion point [resolution/lowerIterators.cpp:1883]
```

So for now, I'm just going to revert the `UnitTest` package and add an issue to rewrite it after the changes to the List API go through.